src/core/server/core_usage_data/core_usage_data_service.ts

```diff
@@ -20,8 +20,9 @@ import type { CoreContext, CoreService } from '@kbn/core-base-server-internal';
 import type { LoggingConfigType } from '@kbn/core-logging-server-internal';
 import type { Logger } from '@kbn/logging';
 import type { HttpConfigType, InternalHttpServiceSetup } from '@kbn/core-http-server-internal';
+import type { ElasticsearchServiceStart } from '@kbn/core-elasticsearch-server';
+import type { ElasticsearchConfigType } from '@kbn/core-elasticsearch-server-internal';
```

Comment:
> It's maybe too early to say, but how should we approach the need for sometimes having to use internal core domain types outside of core?

The initial idea was to simply forbid import from `-internal` package by consumers outside of Core via an ESlint rule. Note that now that we progressed quite a bit in the migration, we can say that this will cause problem given we saw that some 'internal' stuff is (at least currently) exported from these packages by our entrypoints and consumed externally. 

One option could be to extract those parts to dedicated 'util' packages, or maybe have a way to flag things that are allowed to import from internal packages. I'm not sure yet to be honest. We definitely need to think about it once we complete the rest of the migration.